pkg/controller/controller_utils.go

```diff
@@ -483,6 +483,9 @@ func (r RealPodControl) CreatePodsWithControllerRef(namespace string, template *
 	if controllerRef.Controller == nil || *controllerRef.Controller != true {
 		return fmt.Errorf("controllerRef.Controller is not set")
 	}
+	if controllerRef.BlockOwnerDeletion == nil || *controllerRef.BlockOwnerDeletion != true {
```

Comment:
> I am confused about why we have this function

It's a util function shared by multiple controllers. We can't guarantee every author of controllers will test the controllerRef, so we make a check here.

> I am also confused about why it's so important that controllers set the block owner deletion field.

User will expect controllees to block controller's deletion during foreground deletion, so a controller should set the blockOwnerDeletion field.